packages/main/src/mainWindow.ts

```diff
@@ -101,10 +103,10 @@ async function createWindow() {
   });
 
   browserWindow.on('close', e => {
-    e.preventDefault();
-    if (isLinux) {
-      browserWindow.minimize();
-    } else {
+    const closeBehaviorConfiguration = configurationRegistry.getConfiguration('preferences');
+    const minimizeonclose = closeBehaviorConfiguration.get<boolean>('minimizeonclose');
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

It's because configuration registry may not be defined there

Reply:
also if configuration is not defined we shouldn't try to get the property as well

Reply:
You should probably declare a variable `configurationRegistry` at an upper scope and remove the const in the initialization after ready state is reached

Reply:
I am trying to declare it global, but it seems like I keep getting the value undefined in the actual onClose call.

Reply:
It turns out, this property is always ready by the time the window is open. Not sure if this is needed?